boards/BMS/Src/Io/Io_LTC6813.c

```diff
@@ -0,0 +1,368 @@
+    tx_cmd[2] = (uint8_t)(tx_cmd_pec15 >> 8);
+    tx_cmd[3] = (uint8_t)tx_cmd_pec15;
+
+    return (Io_SharedSpi_Transmit(ltc_6813.spi, tx_cmd, NUM_OF_CMD_BYTES) ==
+            HAL_OK)
+               ? EXIT_CODE_OK
+               : EXIT_CODE_UNIMPLEMENTED;
+}
+
+static ExitCode Io_LTC6813_PollAdcConversion(void)
+{
+    uint8_t tx_cmd[NUM_OF_CMD_BYTES];
+    uint8_t rx_data = 0xFF;
+
+    tx_cmd[0] = (uint8_t)PLADC;
+    tx_cmd[1] = (uint8_t)(PLADC >> 8);
+
+    uint16_t tx_cmd_pec15 =
+        Io_CalculatePec15(tx_cmd, NUM_OF_PEC15_BYTES_PER_CMD);
+    tx_cmd[2] = (uint8_t)(tx_cmd_pec15 >> 8);
+    tx_cmd[3] = (uint8_t)tx_cmd_pec15;
+
+    uint32_t adc_conversion_timeout_counter = 0U;
+
+    while (rx_data == 0xFF)
+    {
+        if (Io_SharedSpi_TransmitAndReceive(
+                ltc_6813.spi, tx_cmd, NUM_OF_CMD_BYTES, &rx_data, 1U) != HAL_OK)
+        {
+            return EXIT_CODE_UNIMPLEMENTED;
+        }
+
+        ++adc_conversion_timeout_counter;
+
+        // Timeout counter threshold of 10 was chosen arbitrarily.
+        if (adc_conversion_timeout_counter >= 10U)
+        {
+            return EXIT_CODE_TIMEOUT;
+        }
+    }
+
+    return EXIT_CODE_OK;
+}
+
+static void Io_LTC6813_ParseCellsAndPerformPec15Check(
+    size_t   current_ic,
+    size_t   current_register_group,
+    uint8_t *rx_cell_voltages)
+{
+    size_t cell_voltage_index = current_ic * NUM_OF_CELL_VOLTAGE_RX_BYTES;
+
+    for (size_t current_cell = 0U;
+         current_cell < NUM_OF_CELLS_PER_LTC6813_REGISTER_GROUP; current_cell++)
+    {
+        uint32_t cell_voltage =
+            (uint32_t)(rx_cell_voltages[cell_voltage_index]) |
+            (uint32_t)((rx_cell_voltages[cell_voltage_index + 1] << 8));
+
+        ltc_6813.cell_voltages[current_ic]
+                              [current_cell +
+                               current_register_group *
+                                   NUM_OF_CELLS_PER_LTC6813_REGISTER_GROUP] =
+            (uint16_t)cell_voltage;
+
+        // Each cell voltage is represented as a pair of bytes. Therefore
+        // increment the cell voltage index by 2 to retrieve the next cell
+        // voltage.
+        cell_voltage_index += 2U;
+    }
+
+    uint32_t received_pec15 =
+        (uint32_t)(rx_cell_voltages[cell_voltage_index] << 8) |
+        (uint32_t)(rx_cell_voltages[cell_voltage_index + 1]);
+    uint32_t calculated_pec15 = Io_CalculatePec15(
+        &rx_cell_voltages[current_ic * NUM_OF_CELL_VOLTAGE_RX_BYTES], 6U);
+
+    if (received_pec15 != calculated_pec15)
+    {
+        // Increment the number of PEC15 errors.
+        ltc_6813.pec15_error_counter++;
+    }
+}
+
+void Io_LTC6813_Init(
+    SPI_HandleTypeDef *hspi,
+    GPIO_TypeDef *     nss_port,
+    uint16_t           nss_pin)
+{
+    assert(hspi != NULL);
+
+    ltc_6813.spi                 = Io_SharedSpi_Create(hspi, nss_port, nss_pin);
+    ltc_6813.pec15_error_counter = 0U;
+    memset(
```

Comment:
I remember Ken telling me its always good practice to initialize the members of the struct. Since the `ltc_6813` struct is static, all values in cell_voltages array should already be set to zero. Is this necessary?

Reply:
I think it's good practice. This way it's a habit to get into and standard across the codebase.